### DISCUSSION TITLE
Hook into the annotation web client function.

### DISCUSSION THREAD

**Discussion by Geeker1** [hypothesis/client#4627] [Ideas]
I need to know if there is a way to hook into the function that creates new annotations with the javascript web client, I'm developing an annotation newsfeed, and i need to add extra information to the annotation object before posting, but i don't want to use the api.

Is that possible ?

**Comment by robertknight** [comment]
Hello @Geeker1 - The Hypothesis client doesn't have a publicly exposed interface for this I'm afraid. Can you provide some examples of the kind of information that you would like to attach to annotations? Also, would this be for web pages where you are embedding the client, or is this functionality you are trying to add to the browser extension or bookmarklet?

**Comment by Geeker1** [comment]
This is for embedding the client in web pages, normally with the api, i can easily create an annotation, get the annotation id and related metadata, then save the data with respect to a specific model on the database.
With the web client, I just highlight the text and annotate, I don't have control over the function that does the annotation, I need this control for filtering the annotations later on.

**Comment by Geeker1** [comment]
I can filter based on urls, tags and timestamps, let's say i want to add a specific tag to an annotation before it is saved, so when i need that annotation later on, i can easily get it, urls might change but i'm not sure tags will.

**Comment by ghost** [comment]
@Geeker1, I found it difficult to follow your comments.

Can you provide a concrete example?  A list of steps of what you're doing in the UI (or would like to do), what the client's behavior is when you do it (as you understand it), and how you like it to work otherwise?  That is, with concrete input.

NB: I don't work at Hypothesis.
